src/runtime/swg-client-event-manager.js

```diff
@@ -0,0 +1,312 @@
+/**
+ * Copyright 2019 The Subscribe with Google Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {AnalyticsEvent,EventOriginator} from '../proto/api_messages';
+
+/**
+ * The callback used to notify registered listeners about new client events.
+ * @callback SwgEventManagerListener
+ * @param {SwgClientEvent} event
+ */
+export let SwgEventManagerListener;
+
+/**
+ * Return false to cancel the event.  You can also modify the event object
+ * which is passed by reference.
+ * @callback SwgEventManagerFilterer
```

Comment:
use function() JSDOC to specify this as well instead of @callback, because that is the pattern in the rest of the code.

Reply:
Also lets use SwgEventFilterer